### DISCUSSION TITLE
Expanding the list of maintainers?

### DISCUSSION THREAD

**Discussion by craighurley** [cloudreach/sceptre#974] [General]
Hello,

Scetpre has been a great tool for me, my colleagues and many others in the community, so first off, a huge thank you for that!

Regulars here notice that work on Sceptre has pretty much halted (life and priorities change -- that's cool; this is not a complaint).  Can we kick off a discussion around expanding the number of maintainers to help lighten the load and keep things progressing?

I am happy to help monitor this repo and related repos for issues and PRs in order to keep things moving.  I'm pretty sure there are others who have a better understanding of the inner workings that would be willing to help out as core maintainers.

What say you @ngfgrant?

**Comment by ykhalyavin** [comment]
We use Sceptre to deploy to 40+ accounts and 5 regions within each account, with 200+ stacks configs inside "config" folder.
Sceptre is invaluable tool to dynamically pass outputs between accounts / regions (e.g. you want to know you "logging" buckets ARNs (that live in logging account) across all accounts).

Other teams in our company use Sceptre to deploy their apps containing lots of services / dependencies to multiple accounts / regions as well.

In other words, it's critical for us that Sceptre is up to date with new python versions, compatible with cfn-lint / etc packages new versions and pull requests are handled in timely manner.

Appreciate your work @ngfgrant you've done on the project, would love to see more people helping out and making the project better!

**Reply by ngfgrant** [reply]
Yeh more than happy to. Iirc there are members of the org who are not in CR that should have merge permissions. If not I am happy to give.

I have been speaking with some devs at Cloudreach. I don’t think sceptre is a priority for them. 

Like the ex-cloudreachers I’ve not been at Cloudreach for well over a year now.

I’d support a move away from CircleCI toward GitHub Actions. 

I’d encourage anyone merging code to remember that this is the sort of tool that you don’t want to cause you any drama.

**Comment by philipg** [comment]
@devenney @njsnx @chipwolf any of you know the core codebase for sceptre and willing to help maintain it?  happy to help chip in with PRs etc but have never looked at the core codebase.

**Reply by ChipWolf** [reply]
Agree; the `master` branch is fairly stale. I have some familiarity with the codebase from previously working at Cloudreach over two years ago.

However, as far as I understand, Sceptre is maintained by their staff as regular BAU activity & it looks like there are several active forks. The demand for additional maintainers should really come from the organization; I can't speak for their appetite, but I'm certain they'd be more open to developers who are using Sceptre in production.

It does look like PRs are being reviewed by other maintainers _(such as @zaro0508)_ as early as 5 hours ago, it'd be useful if one of the maintainers would chip in here.

**Reply by craighurley** [reply]
This is the problem, @zaro0508 is an active contributor along with a few others, but there are no active maintainers around to merge contributions.  There are a number of open PRs in this repo and a desire to add other repos under the sceptre organisation for resources/hooks but nothing is happening.

**Reply by ChipWolf** [reply]
If this is the case; I'd suggest setting up a new GitHub organization, `sceptre-community`,  similar to the `prometheus-community` structure & forking this repo to get some semblance of movement.

**Reply by tiagoasousa** [reply]
Wholeheartedly agree, we need to get some progress on this :)

**Reply by rdkr** [reply]
Hi. I have full access to the current org and can help to progress this. I have just un-stuck the builds and am happy to work with @zaro0508 (and any other maintainers) to get this fully maintainable.

**Reply by craighurley** [reply]
That's great @rdkr , thank you!

While forking was always an option as @ChipWolf said, I'd prefer to see focus remain on this project because it already has a user base and a recognisable name here and in pip.  I think we still need to talk about expanding the list of maintainers to help lighten the load and keep things progressing, in the event that @rdkr falls off radar :)

**Reply by ykhalyavin** [reply]
Thanks @rdkr, can help reviewing existing PRs and creating new ones (we have use cases not supported by current version) -- pretty familiar with the codebase due to workarounds we had to made.

Agree with @craighurley, it would be great to have at least a couple of maintaners who can merge changes in case one disappears.
What do you think @rdkr ?

**Reply by lukeplausin** [reply]
I didn't work much on the core codebase, but I was in the room when Cloudreach founder James Monico announced that Sceptre would go open source. The intention behind their move was to widen participation and to contribute something back to the community. James, Chip, @ngfgrant and myself all left a while ago and it seems that besides the good efforts made by @rdkr there isn't much desire in Cloudreach to continue contributing to the project. I think that expanding the list of maintainers from within the community is the best way to keep the project active and would be in the spirit of the original decision to go open source.

**Comment by zaro0508** [comment]
Hello Sceptre fans. I know many of you have been frustrated by Sceptre’s lack of progress. I would like to give you the good news that @rdkr and I are in the process of trying to revive the infrastructure needed to support our beloved app.  We are getting closer to a state where we can merge code again.  We have gotten the sceptre build and unit tests working again which is why you’ve seen some merged documentation changes lately.  We still have a problem in that we do not have the integration tests running yet.  The integration tests requires an AWS account to run on and we don’t have access to the previously used AWS account anymore.  I think it would be beneficial to the app and broader Sceptre community as a whole if could get the integration tests running before merging code.  For that we will probably need some help from our community.  Please let us know if you have any ideas on how we can get an AWS account for testing Sceptre.  Thanks.

**Reply by tiagoasousa** [reply]
why not migrate to github actions?

**Reply by fabiodouek** [reply]
Hi @zaro0508 , AWS supports many Open Source projects. You can apply for AWS promotional credits, as mentioned here: https://aws.amazon.com/blogs/opensource/aws-promotional-credits-open-source-projects/

**Reply by zaro0508** [reply]
> why not migrate to github actions?
good idea @networkpadawan, contributions are welcome.  however that does not address the needed AWS account.



> Hi @zaro0508 , AWS supports many Open Source projects. You can apply for AWS promotional credits, as mentioned here: https://aws.amazon.com/blogs/opensource/aws-promotional-credits-open-source-projects/

thanks @fabiodouek, hard to tell what the credits cover but will look into it.

**Reply by tiagoasousa** [reply]
@zaro0508 if using githubactions what is the need for a aws account?

**Reply by craighurley** [reply]
> what is the need for a aws account?

To run tests against an aws account.

**Reply by tiagoasousa** [reply]
of course...that makes sense:)

**Reply by zaro0508** [reply]
I tried your suggestion @fabiodouek, submitted application for credits a few weeks ago but didn't get any type of confirmation or reply.  Unfortunately I think that's a dead end.   I'm open to other ideas.
